### PR TITLE
FEAT: add twitter handle

### PR DESCRIPTION
**Pull request by theodorusclarence**
## Description

since you're still using twitter logo on the intent button, so I followed them.

image

CleanShot 2023-10-12 at 21 12 59@2x

## Pre-requisites before merging

- [x] Add `x_username` column to User DB (Text format)
	
	CleanShot 2023-10-12 at 21 16 11@2x

- [x] Update notion template

### CODE REVIEW

File: src/lib/api.ts [hunk 1]
```diff
@@ -100,7 +100,10 @@ export const postAddUser = async (
 
 export const patchUpdateUser = async (
   user: User,
-  param: Pick<UpdateUserArgs, 'name' | 'slug' | 'image' | 'public'>,
+  param: Pick<
```
Comment (mazipan) [inline src/lib/api.ts]: Think about using Omit instead of Pick

File: src/lib/notion.ts [hunk 2]
```diff
@@ -252,7 +253,7 @@ export const submitQuestion = async (param: SubmitQuestionArgs) => {
         title: [
           {
             type: 'text',
-            text: { content: generateNanoId() },
+            text: { content: nanoid() },
```
Comment (mazipan) [inline src/lib/notion.ts]: This reverting previous changes. Is it intended?

Comment (theodorusclarence) [reply src/lib/notion.ts]: whoops, must be because i forget to pull last time

**Comment by mazipan** [comment]
Btw it is okay to just use a normal push instead of force push.
I almost always using squash when merging the PR.

**Comment by theodorusclarence** [comment]
> Btw it is okay to just use a normal push instead of force push. I almost always using squash when merging the PR.

Oh, just a force of habit 😃, I usually always rebase.

Noted

File: src/app/p/[slug]/page.tsx [hunk 3]
```diff
@@ -92,6 +93,18 @@ export default async function PublicPage({
             Tanya ke {owner?.data?.name}
           </h1>
 
+          {owner.data?.x_username && (
```
Comment (mazipan) [inline src/app/p/[slug]/page.tsx]: The owner can be null somehow, how about adding optional chain also?


```suggestion
          {owner?.data?.x_username && (
```

**Comment by mazipan** [comment]
Ah yeah, rebase is good for syncing with the remote.

But when you adding new commit, it is hard for maintainer to see what you have done, since the history can not be tracked.
I can not see what you already change from the previous commit, so I need to re-review again from the beginning.

It is easy for small PR, but it's getting harder when the LOC is bigger.

**Review by mazipan — APPROVED** [review]
Overall okay.
I need to prepare the changes on the Notion first

**Review by mazipan — APPROVED** [review]

**Review by mazipan — APPROVED** [review]

**Pull request merged**